### ISSUE TITLE
Update mollom_entity_?? hooks to use EntityInterface

### ISSUE THREAD

**Issue by YesCT** [YesCT/mollom#1]
Related https://github.com/backdrop/backdrop-issues/issues/2573

Alternative https://github.com/YesCT/mollom/commit/814a250ac1cc35b69a64920f3b13a65d79c0aa10

**Comment by YesCT** [comment]
```
/**
 * Implements hook_entity_insert().
 */
function mollom_entity_insert($entity, $type) {
  list($id) = entity_extract_ids($type, $entity);
  if (!empty($entity->mollom) && !empty($id)) {
    $entity->mollom['id'] = $id;
    $data = (object) $entity->mollom;
    mollom_data_save($data);
  }
}
```

uh... ?? Does this say, if it's a mollom entity, and already exists (has an id), make the mollom id the same as the entity id, and do a mollom save?

when inspecting the $entity object on a save of the test form (mollom-test/form), there was no mollom property in the entity object
![mollom_entity](https://cloud.githubusercontent.com/assets/1970586/23955790/58bd10da-0971-11e7-9b5e-e8c73ea5b9a8.png)

I'm not sure what this might look like.

Maybe
```
/**
 * Implements hook_entity_insert().
 */
function mollom_entity_insert(EntityInterface $entity) {
  if ($entity->entityType() == 'mollom') && $entity->id())) {
    $data = (object) .... // build the data. Need to figure out its structure.
    mollom_data_save($data);
    // or convert more things to use EntityInterface and
    mollom_data_save($entity);
  }
}
```

**Comment by jenlampton** [comment]
I'm not sure this function needs to change at all?
https://api.backdropcms.org/api/backdrop/core%21modules%21entity%21entity.api.php/function/hook_entity_insert/1

**Comment by YesCT** [comment]
Yeah, I'm not sure either. Changing the test to not use module_invoke_all would get around the fail from entity hooks being invoked with two different signatures: https://github.com/YesCT/mollom/commit/814a250ac1cc35b69a64920f3b13a65d79c0aa10
